components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,357 @@
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+- (void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  
+  CGRect frame = self.view.frame;
+  frame.size = size;
+  self.view.frame = frame;
+  CGRect headerFrame = _header.frame;
+  headerFrame.size.width = size.width;
+  _header.frame = headerFrame;
+
+  CGRect tableFrame = _tableViewController.tableView.frame;
+  tableFrame.size.width = size.width;
+  _tableViewController.tableView.frame = tableFrame;
+  [_tableViewController.tableView reloadData];
+  [_tableViewController.tableView setNeedsLayout];
+  /// We need this call to `layoutIfNeeded` to get the correct contentSize for the table.
+  [_tableViewController.tableView layoutIfNeeded];
+  CGFloat tableHeight = _tableViewController.tableView.contentSize.height;
+  CGFloat height = CGRectGetHeight(headerFrame) + tableHeight;
+  CGSize updatedSize = CGSizeMake(size.width, height);
+  self.preferredContentSize = updatedSize;
+}
+
+#pragma mark - Table view delegate
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+  MDCActionSheetAction *action = self.actions[indexPath.row];
+
+  [self.presentingViewController dismissViewControllerAnimated:YES completion:^(void){
+    if (action.completionHandler) {
+      action.completionHandler(action);
+    }
+  }];
+}
+
+- (void)setTitle:(NSString *)title {
+  _actionSheetTitle = title;
+  _header.title = title;
```

Comment:
Ensuring that our properties are mapped to our header instance will prove to be somewhat challenging over time, so it may less brittle to simply create the `_header` instance in the init method. This way we can just use the header view to store and retrieve all of our customizable properties.

Reply:
Done